.github/workflows/pull-contracts.yml

```diff
@@ -0,0 +1,102 @@
+name: Pull Contracts
+
+on:
+  pull_request_target:
+    types: 
+      - opened
+      - synchronize
+      - reopened
+      - edited
+    branches:
+      - main
+    paths:
+      - ./contracts/**/*
+
+# ! danger !:
+# https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections
+jobs:
+  Validate:
+    # TODO: if: name PR starts with ...
+    runs-on: ubuntu-18.04
+    steps:
+      # checkout repo/branch being pushed to
+      # https://github.com/actions/checkout
+      - name: "base: checkout"
+        uses: actions/checkout@v2
+        with:
+          # owner / repo (EthVM/evm-soruce-verification)
+          repository: ${{ github.event.pull_request.base.repo.full_name }}
+          # branch (main)
+          ref: ${{ github.event.pull_request.base.ref }}
+
+      - name: install NodeJS
+        uses: actions/setup-node@v2
+        with:
+          # cache npm dependencies
+          # https://github.com/actions/setup-node#caching-packages-dependencies
+          cache: 'npm'
+          node-version: 16.13.2
+          registry-url: "https://registry.npmjs.org/"
+
+      - name: set NPM version
+        run: npm i -g npm@6.14.16
+
+      - name: "create .env file"
+        run: |
+          cp .env.example .env
+          sed -i'' "s/^GITHUB_TOKEN=/GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}/" .env
```

Comment:
you can set the env vars like this 

```
env:
          First_Name: Mona
```
https://docs.github.com/en/actions/learn-github-actions/environment-variables